src/wrapper/errors.rs

```diff
@@ -31,25 +31,31 @@ pub enum Error {
     TryLock,
     #[error("JavaVM null method pointer for {0}")]
     JavaVMMethodNotFound(&'static str),
+    #[error("Field already set: {0}")]
+    FieldAlreadySet(String),
+    #[error("Throw failed with error code {0}")]
+    ThrowFailed(i32),
+    #[error("Parse failed for input: {1}")]
+    ParseFailed(#[source] combine::error::StringStreamError, String),
+    #[error("JNI call failed")]
+    JniCall(#[source] JniError),
+}
+
+#[derive(Debug, Error)]
+pub enum JniError {
     #[error("Unknown error")]
     Unknown,
-    #[error("Current thread is not attached to the Java VM")]
+    #[error("Thread was detached from the VM")]
```

Comment:
(nit) The previous version seems to be more precise: IIUC, this error will happen if you somehow keep JNIEnv around after the nativa thread was detached and then try to use any JNIEnv in a detached thread.

Reply:
Oh right, the previous message is more descriptive. Let me roll that back before you merge it.

Reply:
Thanks!